python/lsst/pipe/tasks/measurementDriver.py

```diff
@@ -0,0 +1,690 @@
+    ...     butler, ["g", "r", "i"], "deepCoadd_calexp", ...
+    ... )
+    >>> driver = MultiBandMeasurementDriverTask(config=config)
+    >>> catalog = driver.run(mExposure, "r")
+    >>> catalog.writeFits("meas_catalog.fits")
+    """
+
+    ConfigClass = MultiBandMeasurementDriverConfig
+    _DefaultName = "multiBandMeasurementDriver"
+    _Deblender = "scarlet"
+
+    def run(
+        self,
+        mExposure: afwImage.MultibandExposure | list[afwImage.Exposure],
+        band: str | None = None,
+        bands: list[str] | None = None,
+        catalog: afwTable.SourceCatalog = None,
+        idGenerator: measBase.IdGenerator = None,
+    ) -> afwTable.SourceCatalog:
+        """
+        Process a multi-band exposure or a list of exposures.
+
+        Parameters
+        ----------
+        mExposure :
+            Multi-band data. May be a single `MultibandExposure` or a list of
+            exposures associated with different bands in which case ``bands``
+            must be provided.
+        band : optional
+            Reference band to use for detection and measurement.
+        bands : optional
+            List of bands associated with the exposures in ``mExposure``. Only
+            required if ``mExposure`` is a list of single-band exposures.
+        catalog : optional
+            Catalog to be extended by the driver task. If not provided, a new
+            catalog will be created and populated.
+        idGenerator : optional
+            Generator for unique source IDs.
+
+        Returns
+        -------
+        catalog :
+            Catalog containing the measured sources.
+        """
+
+        # Basic sanity checks to ensure the inputs are consistent.
+        if (band is None) != (bands is None):
+            raise ValueError("'band' and 'bands' must be provided together or not at all.")
+        if band is not None and bands is not None:
+            if band not in bands:
+                raise ValueError(f"Reference band '{band}' is not in the list of bands: {bands}")
+
+        # Store the reference band for later use.
+        self.band = band
+
+        # Convert mExposure to a MultibandExposure object if not already in
+        # that form. Save the result as an instance attribute for later use.
+        self.mExposure = self._buildMultibandExposure(mExposure, bands)
+
+        if self.band not in self.mExposure:
+            raise ValueError(f"Requested band '{band}' is not present in the multiband exposure.")
+
+        # We use a reference band for band-specific tasks like detection and
+        # measurement.
+        self.exposure = self.mExposure[self.band]
+        self.log.info(f"Using '{self.band}' band as the reference band for band-specific tasks")
+
+        self.catalog = catalog
+        self.idGenerator = idGenerator
+
+        return super().run()
+
+    def _deblendSources(self):
+        self.catalog, modelData = self.deblend.run(mExposure=self.mExposure, mergedSources=self.catalog)
```

Comment:
modelData should be added to the struct that is returned, as it contains important information about blends that is often useful for debugging.